tests/rptest/clients/rpk.py

```diff
@@ -90,7 +90,7 @@ def produce(self,
         ]
         if headers:
             cmd += ['-H ' + h for h in headers]
-        if partition:
+        if partition is not None:
```

Comment:
nice catch. ugh _falsy_ is my least favorite language feature. ok maybe not the least favorite